python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -975,9 +988,74 @@ def do_evaluate(
             child.evaluate(df, context=context, mapping=mapping)
             for child in self.children
         ]
-        if self.name == pl_expr.TemporalFunction.Year:
+        if self.name in TemporalFunction._temporal_extraction_function_map:
             (column,) = columns
-            return Column(plc.datetime.extract_year(column.obj))
+            if self.name == pl_expr.TemporalFunction.Millisecond:
+                return Column(
+                    plc.datetime.extract_datetime_component(column.obj, "millisecond")
+                )
+            if self.name == pl_expr.TemporalFunction.Microsecond:
+                milis = plc.datetime.extract_datetime_component(
+                    column.obj, "millisecond"
+                )
+                micros = plc.datetime.extract_datetime_component(
+                    column.obj, "microsecond"
+                )
+                processed_mili = plc.binaryop.binary_operation(
+                    milis,
+                    plc.interop.from_arrow(pa.scalar(1_000, type=pa.int64())),
+                    plc.binaryop.BinaryOperator.MUL,
+                    plc.types.DataType(plc.types.TypeId.INT64),
+                )
+                total_micros = plc.binaryop.binary_operation(
+                    micros,
+                    processed_mili,
+                    plc.binaryop.BinaryOperator.ADD,
+                    plc.types.DataType(plc.types.TypeId.INT64),
```

Comment:
*issue*: `self.dtype` contains the correct output dtype. It is Int32, not Int64. We also don't _need_ to compute in int64 here. There are only $10^6$ possible answers here.

Reply:
This should be resolved, I think we do all computations at the minimum lossless width now.